modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/metadata/MetadataExtensionFunctionalTestCase.java

```diff
@@ -230,6 +249,27 @@ protected void assertExpectedType(Typed typedModel, MetadataType expectedType, b
     assertThat(typedModel.hasDynamicType(), is(isDynamic));
   }
 
+  protected <T extends ComponentModel> void assertResolvedKey(MetadataResult<ComponentMetadataDescriptor<T>> result,
+                                                              MetadataKey metadataKey) {
+    assertThat(result.get().getMetadataAttributes().getKey().isPresent(), is(true));
+    MetadataKey resultKey = result.get().getMetadataAttributes().getKey().get();
+    assertSameKey(metadataKey, resultKey);
+
+    MetadataKey child = metadataKey.getChilds().stream().findFirst().orElseGet(() -> null);
+    MetadataKey otherChild = resultKey.getChilds().stream().findFirst().orElseGet(() -> null);
+    while (child != null && otherChild != null) {
```

Comment:
All this should not be necessary in order to compare if two keys are the same. 
If current `equals` doesn't work like this, we should update it.